fenicsadapter/fenicsadapter.py

```diff
@@ -327,36 +285,16 @@ def initialize(self, coupling_subdomain, mesh, function_space, dimensions=2):
         self._read_function_type = determine_function_type(function_space)
         self._function_space = function_space
 
-        return self._interface.initialize()
+        precice_dt = self._interface.initialize()
 
-    def initialize_data(self, write_function=None):
-        """
-        Set non-standard values as initial conditions to coupling problem
-
-        Parameters
-        ----------
-        write_function : Object of class dolfin.functions.function.Function
-            FEniCS function related to the quantity to be written by FEniCS during each coupling iteration.
-
-        Returns
-        -------
-        read_data = array_like
-            Numpy array containing the read data.
-        """
-        self._non_standard_initialization = True
-
-        if write_function:
-            if self._interface.is_action_required(precice.action_write_initial_data()):
-                self.write_data(write_function)
-                self._interface.mark_action_fulfilled(precice.action_write_initial_data())
+        if self._interface.is_action_required(precice.action_write_initial_data()):
+            assert(write_function)  # if action is required, write function MUST be given.
+            self.write_data(write_function)
+            self._interface.mark_action_fulfilled(precice.action_write_initial_data())
 
         self._interface.initialize_data()
```

Comment:
Making `initialize_data()` a part of `initialize()` means it will be called even if the user wishes to start with all coupling variables equal to zero. Is this acceptable use of the preCICE API?

Reply:
Resolved in discussion.